nearcore/src/shard_tracker.rs

```diff
@@ -135,15 +157,21 @@ impl ShardTracker {
     }
 
     fn update_epoch(&mut self, block_hash: &CryptoHash) -> Result<(), EpochError> {
-        let epoch_id = {
+        let (_, epoch_id) = {
             let mut epoch_manager = self.epoch_manager.write().expect(POISONED_LOCK_ERR);
-            epoch_manager.get_epoch_id(block_hash)?
+            (epoch_manager.get_prev_epoch_id(block_hash)?, epoch_manager.get_epoch_id(block_hash)?)
         };
         if self.current_epoch_id != epoch_id {
+            //assert_eq!(self.current_epoch_id, prev_epoch_id);
```

Comment:
should this be removed?

Reply:
Yeah, I wasn't sure because i had the confusion for shard_tracker unstrack_shards